x-pack/plugin/esql/qa/testFixtures/src/main/resources/ip.csv-spec

```diff
@@ -163,12 +163,11 @@ eth1           |beta           |127.0.0.1
 eth1           |beta           |127.0.0.1                                                                        |128.0.0.1                |128.0.0.1
 lo0            |gamma          |fe80::cae2:65ff:fece:feb9                                                        |fe81::cae2:65ff:fece:feb9|fe81::cae2:65ff:fece:feb9
 eth0           |gamma          |fe80::cae2:65ff:fece:feb9                                                        |127.0.0.3                |127.0.0.3
-eth0           |epsilon        |[fe80::cae2:65ff:fece:feb9, fe80::cae2:65ff:fece:fec0, fe80::cae2:65ff:fece:fec1]|fe80::cae2:65ff:fece:fec1|fe80::cae2:65ff:fece:fec1
```

Comment:
What's the behavior change here? Do we want that?

Reply:
If there are multi-valued expressions in the inlist, the test in string.csv-spec returns null, and have a warning says "single-value function encountered multi-value". This test case also has this warning but it doesn't return a null on main, perhaps the old/current implementation handles them(ip and string) differently. I'm trying to make them behave the same.

Reply:
Oh, interesting! I think @bpintea wrote IN the first time around. @bpintea do you know which behavior we want in all cases?

Reply:
IMO this line should stay: `IN` is syntactic sugar for `==` _and_ `OR`. So the presence of a single MV element in IN's list should not nullify the result: we optimise statements like `a == b OR a == c` as `a IN (b, c)` and the non-IN expression can return true if one equality holds, even when the other is `null` due to comparison against a MV. (Genearlly, this is the case with other functions too, such as `CASE(true, 1, [1, 2])` returning `1`, one MV argument in its argument list doesn't automatically mean a `null`).
Besides that, we need to keep the 3VL: (true, null) -> true, (false, null) -> null.

Reply:
> IMO this line should stay: `IN` is syntactic sugar for `==` _and_ `OR`. So the presence of a single MV element in IN's list should not nullify the result: we optimise statements like `a == b OR a == c` as `a IN (b, c)` and the non-IN expression can return true if one equality holds, even when the other is `null` due to comparison against a MV. (Genearlly, this is the case with other functions too, such as `CASE(true, 1, [1, 2])` returning `1`, one MV argument in its argument list doesn't automatically mean a `null`). Besides that, we need to keep the 3VL: (true, null) -> true, (false, null) -> null.

Thanks for the explanation! I'll allow MV in the inlist, and also keep the 3VL rule.